apisix/plugins/echo.lua

```diff
@@ -70,4 +92,21 @@ function _M.body_filter(conf, ctx)
     ngx.arg[2] = true
 end
 
+function _M.header_filter(conf, ctx)
+    if conf.headers_arr then
+        local field_cnt = #conf.headers_arr
+        for i = 1, field_cnt, 2 do
+            ngx.header[conf.headers_arr[i]] = conf.headers_arr[i+1]
+        end
+    end
+end
+
+function _M.access(conf, ctx)
+    local auth_header = ngx.headers["Authorization"]
+    if not auth_header then
+        return 401, { message = conf.unathorized_body }
+    end
+    return 200, { message = conf.authorized_body }
```

Comment:
Can the value corresponding to the auth header be verified?

Reply:
@agile6v  you mean like we do in the basic-auth plugin extract the values and check ?